---
title: "class9- Alpha fold"
format: html
author: Monika Ramos
---

**skip section 1 **

**The important role of water**

Q4: Water molecules normally have 3 atoms. Why do we see just one atom per water molecule in this structure?
Given that hydrogen is the smallest atom, it is not easy to see here.

Q5: There is a critical “conserved” water molecule in the binding site. Can you identify this water molecule? What residue number does this water molecule have

308

Q6: Generate and save a figure clearly showing the two distinct chains of HIV-protease along with the ligand. You might also consider showing the catalytic residues ASP 25 in each chain and the critical water (we recommend “Ball & Stick” for these side-chains). Add this figure to your Quarto document.

![](1HSG.png)


#Viewing PDB structures with Molstart

![](1HSG.png)
**3. Introduction to Bio3D in R**
Bio3D is an R package for structural bioinformatics. Features include the ability to read, write and analyze biomolecular structure, sequence and dynamic trajectory data.
```{r}
library(bio3d)
```

Reading PDB file data into R
```{r}
pdb <- read.pdb("1hsg")
pdb
```
Q7: How many amino acid residues are there in this pdb object?
There are 128 amino acid residues
Q8: Name one of the two non-protein residues? 
MK1 
Q9: How many protein chains are in this structure? 
2
```{r}
attributes(pdb)
```
```{r}
head(pdb$atom)
```
**Predicting functional motions of a single structure**

```{r}
adk <- read.pdb("6s36")
adk
```


Normal mode analysis (NMA) it is a bioinformatics method for predicting functional motions. It will show us the parts of the protein that are "flexible" (ie. most dynamic).
Perform flexiblity prediction

```{r}
m <- nma(adk)
plot(m)
```

Make a movie of this moving by creating a trajectory
```{r}
mktrj(m, file = 'adk_nma.pdb')
```

** 4. Comparative structure analysis of all Adenylate Kinase (ADK) structures
The goal of this section is to perform principal component analysis (PCA) on the complete collection of Adenylate kinase structures in the protein data-bank (PDB) ** 
    ADK  --> PDB database --> homologous PDB
get.seq() --> blast.pdb() --> get.pdb()
```{r}
# Install packages in the R console NOT your Rmd/Quarto file

#install.packages("bio3d")
#install.packages("devtools")
#install.packages("BiocManager")

#BiocManager::install("msa")
devtools::install_bitbucket("Grantlab/bio3d-view")
```

Q10. Which of the packages above is found only on BioConductor and not CRAN? 
msa

Q11. Which of the above packages is not found on BioConductor or CRAN?
 bio3d.view
 
Q12. True or False? Functions from the devtools package can be used to install packages from GitHub and BitBucket? TRUE

First we get the sequence of ADK and use this to search the PDB databse.
```{r}
library(bio3d)
aa <- get.seq("1ake_a")
aa
```
Q13. How many amino acids are in this sequence, i.e. how long is this sequence? 

214 amino acids 

```{r}
hits <- NULL
hits$pdb.id <- c('1AKE_A','6S36_A','6RZE_A','3HPR_A','1E4V_A','5EJE_A','1E4Y_A','3X2S_A','6HAP_A','6HAM_A','4K46_A','3GMT_A','4PZL_A')
```


```{r}
pdb.annotate(hits$pdb.id)
#Download related files
files <- get.pdb(hits$pdb, path = "pdbs", split=T, gzip = T)
```

Align and superpose structures
```{r}
# Align releated PDBs
pdbs <- pdbaln(files, fit = TRUE, exefile="msa")
```

```{r}
# Vector containing PDB codes for figure axis
ids <- basename.pdb(pdbs$id)

# Draw schematic alignment
plot(pdbs, labels=ids)
```

Annotate collected PDB structures
```{r}
anno <- pdb.annotate(ids)
unique(anno$source)
anno
```
Principal component analysis
```{r}
# Perform PCA
pc.xray <- pca(pdbs)
plot(pc.xray)
```
```{r}
# Calculate RMSD
rd <- rmsd(pdbs)

# Structure-based clustering
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)

plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
```

